checker/src/main/java/org/checkerframework/checker/index/IndexMethodIdentifier.java

```diff
@@ -37,6 +38,9 @@ public class IndexMethodIdentifier {
   /** The LengthOf.value argument/element. */
   private final ExecutableElement lengthOfValueElement;
 
+  /** The {@code java.lang.String#indexOf} methods that take a string argument. */
```

Comment:
I would add "as a non-receiver argument" because I think all of these also take a string as the receiver.

Also, please add "and String#lastIndexOf".